app/src/org/commcare/dalvik/activities/CommCareHomeActivity.java

```diff
@@ -1405,37 +1532,86 @@ private void refreshView() throws SessionUnavailableException{
-            viewOldForms.setVisibility(Button.VISIBLE);
+            if(isUsingNewUI()){
+                adapter.setButtonVisibility(R.layout.home_savedforms_button, false, false);
+            } else {
+                viewOldForms.setVisibility(Button.VISIBLE);
+            }
         }
 
-        
-        View formRecordPane = this.findViewById(R.id.home_formspanel);
-        
-        if((!CommCarePreferences.isIncompleteFormsEnabled() && !CommCarePreferences.isSavedFormsEnabled())) {
-            formRecordPane.setVisibility(View.GONE);
+
+        if(isUsingNewUI()){
+            // set adapter to hide the buttons...
+            boolean showSavedForms = CommCarePreferences.isSavedFormsEnabled();
+            boolean showIncompleteForms = CommCarePreferences.isIncompleteFormsEnabled();
+
+            Log.i("ShowForms","ShowSavedForms: " + showSavedForms + " | ShowIncompleteForms: " + showIncompleteForms);
+
+            adapter.setButtonVisibility(R.layout.home_savedforms_button, false, !showSavedForms);
+            adapter.setButtonVisibility(R.layout.home_incompleteforms_button, false, !showIncompleteForms);
+
+            adapter.notifyDataSetChanged();
         } else {
+            View formRecordPane = this.findViewById(R.id.home_formspanel);
+
+            if ((!CommCarePreferences.isIncompleteFormsEnabled() && !CommCarePreferences.isSavedFormsEnabled())) {
+                formRecordPane.setVisibility(View.GONE);
+            } else {
             
             /*
              * Not in sense mode
              * Form records are visible unless specifically set to be on/off
              */
-            
-            formRecordPane.setVisibility(View.VISIBLE);
-            
-            if(!CommCarePreferences.isSavedFormsEnabled()){
-                viewOldForms.setVisibility(View.GONE);
-            } else {
-                viewOldForms.setVisibility(View.VISIBLE);
+
+                formRecordPane.setVisibility(View.VISIBLE);
+
+                if (!CommCarePreferences.isSavedFormsEnabled()) {
+                    if(viewOldForms != null) viewOldForms.setVisibility(View.GONE);
+                } else {
+                    if(viewOldForms != null) viewOldForms.setVisibility(View.VISIBLE);
+                }
+
+                if (!CommCarePreferences.isIncompleteFormsEnabled()) {
+                    if(viewIncomplete != null) viewIncomplete.setVisibility(View.GONE);
+                } else {
+                    if(viewIncomplete != null) viewIncomplete.setVisibility(View.VISIBLE);
+                }
+
             }
-            
-            if(!CommCarePreferences.isIncompleteFormsEnabled()){
-                viewIncomplete.setVisibility(View.GONE);
+        }
+    }
+
+    private void setSyncText(Pair<Long, int[]> syncDetails, String syncKey) {
+        String key = syncKey;
+        if(key == null){
+            key = isDemoUser() ? "home.sync.demo" : "home.sync";
```

Comment:
There shouldn't be a need to introduce a new 'key' variable. You can just overwrite the 'syncKey' variable, since strings are immutable and hence won't mutate any external state.
I prefer this because it introduces less indirection in terms of variables, though I'm definitely open to a different style if you prefer something else.